transient/configuration.py

```diff
@@ -0,0 +1,331 @@
+    """
+
+    force = fields.Bool(missing=False)
+
+
+class _TransientListConfigSchema(_TransientConfigSchema):
+    """Defines the schema for the Transient-list configuration and validates
+       the fields during deserialization
+
+       Note that this class is a wrapper to maintain symmetry with the other
+       schemas.
+    """
+
+    pass
+
+
+class _TransientRunConfigSchema(_TransientConfigSchema):
+    """Defines the schema for the Transient-run configuration and validates the
+       fields during deserialization
+    """
+
+    config = fields.Str(allow_none=True)
+    copy_in_before = fields.List(fields.Str(), missing=[])
+    copy_out_after = fields.List(fields.Str(), missing=[])
+    copy_timeout = fields.Int(allow_none=True)
+    prepare_only = fields.Bool(missing=False)
+    qemu_args = fields.List(fields.Str(), allow_none=True)
+    qmp_timeout = fields.Int(missing=10)
+    shutdown_timeout = fields.Int(missing=20)
+    ssh_command = fields.Str(allow_none=True)
+    ssh_bin_name = fields.Str(missing="ssh")
+    ssh_port = fields.Int(allow_none=True)
+    ssh_timeout = fields.Int(missing=90)
+    ssh_user = fields.Str(missing="vagrant")
+    ssh_console = fields.Bool(missing=False)
+    ssh_with_serial = fields.Bool(missing=False)
+    shared_folder = fields.List(fields.Str(), missing=[])
+
+
+def _option_was_set_in_cli(option: Any) -> bool:
+    """Returns True if an option was set in the command line
+    """
+    if (
+        option == IntegerNotGiven
+        or option == StringNotGiven
+        or option == ListNotGiven
+        or option == TupleNotGiven
+        or option == False
+    ):
+        return False
+
+    return True
+
+
+def _option_was_not_set_in_cli(option: Any) -> bool:
+    """Returns True if an option was NOT set in the command line
+    """
+    return not _option_was_set_in_cli(option)
+
+
+def _get_line_number_of_option_in_config_file(option: str, config_file_path: str) -> int:
+    """Returns the line number where the option is found in the config file
+    """
+    with open(config_file_path) as config_file:
+        for line_number, line in enumerate(config_file, start=1):
+            if option in line:
+                return line_number
+
+    return -1
+
+
+def _print_error_message_for_invalid_config_file_option(
+    error: ValidationError, config_file_path: str
+) -> None:
+    """Prints an error message regarding an invalid configuration in the config file
+       and its associated line number
+    """
+    for invalid_option in error.messages:
+
+        # Revert the option to its preformatted state
+        invalid_option = invalid_option.replace("_", "-")
+
+        line_number = _get_line_number_of_option_in_config_file(
+            invalid_option, config_file_path
+        )
+
+        if line_number != -1:
+            print(
+                f"Invalid option on line {line_number} in configuration file: {invalid_option}",
+                file=sys.stderr,
+            )
```

Comment:
I just copied existing logic in `cli.py`:

```
print("No images match selection", file=sys.stderr)
```

Should I change these to use `logging.error` as well?

Reply:
No, cli.py can print to stderr, as it would only be used by someone using the command line. Things outside of that module should be written as though they could be used from a lib, so avoid writing to stdout/stderr when possible

Reply:
Understood. I implemented your request.